### ISSUE TITLE
Continue pre-training on specific corpus

### ISSUE THREAD

**Issue by 1029694141** [kb-labb/kb_bart#1]
Hi,

First thanks for the work on this repo !

Is this repo support continue pre-training on huggingface：（facebook/bart-base） pretrained model ？

**Comment by Lauler** [comment]
I don't know if it does because we haven't tried. Technically it should work if you have the original BART fairseq checkpoints from Facebook.

I don't think that the Huggingface version of the model is a proper checkpoint that has optimizer states and learning rate schedule saved. So you cannot just resume training from Huggingface model file (without faking optimizer states and LR schedule). Also, when converting the model to Huggingface compatible format, it is common to rename some layers and make minor [adjustments](https://github.com/kb-labb/kb_bart/blob/main/save_to_huggingface.py). I think you would have to reverse these to be able to continue training the Huggingface model with fairseq. 

I recommend that you instead first try resuming pretraining from these pre-trained models from Fairseq: https://github.com/facebookresearch/fairseq/blob/main/examples/bart/README.md#pre-trained-models . Look at the [`checkpoint` args for fairseq](https://fairseq.readthedocs.io/en/latest/command_line_tools.html#checkpoint) on how to point to the chosen checkpoint file for pretraining. You should change `--save-dir` and `--restore-file` to point to the `model.pt` provided by Facebook. 

I'm not sure which vocabulary file they used, but I think it's the same `dict.txt` as RoBERTA. 

```wget -O gpt2_bpe/dict.txt https://dl.fbaipublicfiles.com/fairseq/gpt2_bpe/dict.txt```
